autotest/test_external_models.py

```diff
@@ -0,0 +1,134 @@
+import os
+from pathlib import Path
+from shutil import copytree
+
+import flopy
+import pytest
+from compare import (
+    Comparison,
+    detect_comparison,
+    setup_comparison,
+    setup_mf5to6,
+    setup_simulation,
+)
+from framework import TestFramework
+from modflow_devtools.models import DEFAULT_REGISTRY, LocalRegistry
+
+PREFIXES = ["mf6/test", "mf6/large", "mf2005"]
+EXCLUDE = {
+    "alt_model",
+    "test205_gwtbuy-henrytidal",
+    # todo reinstate after 6.5.0 release
+    "test001d_Tnewton",
+    # remove tests with nwt usg conductance weighting
+    "test006_gwf3_gnc_nr_dev",
+    "test006_gwf3_nr_dev",
+    "test014_NWTP3High_dev",
+    "test015_KeatingLike_disu_dev",
+    "test041_flowdivert_nr_dev",
+    "test016_Keating_disu_dev",
+    "test053_npf-a-nwt_dev",
+    "test053_npf-b-nwt_dev",
+    # todo reinstate after resolving convergence failure
+    "test014_NWTP3Low_dev",
+    "test1002_biscqtg_disv_gnc_nr_dev",
+    "test1002_biscqtg_disv_nr_MD_dev",
+    "test1002_biscqtg_disv_nr_RCM_dev",
+    "test1002_biscqtg_disv_nr_dev",
+    "testWetDry/mf2005",
+}
+
+
+def pytest_generate_tests(metafunc):
+    if "model_name" in metafunc.fixturenames:
+        models_paths = metafunc.config.getoption("--models-path")
+        models_paths = [
+            Path(p).expanduser().resolve().absolute() for p in models_paths or []
+        ]
+        registry = LocalRegistry() if any(models_paths) else DEFAULT_REGISTRY
+        registry_type = type(registry).__name__.lower().replace("registry", "")
+        metafunc.parametrize("registry", [registry], ids=[registry_type])
+        models = []
+        if "local" in registry_type:
+            namefile_pattern = (
+                metafunc.config.getoption("--namefile-pattern") or "mfsim.nam"
+            )
+            for path in models_paths:
+                registry.index(path, namefile=namefile_pattern)
+            models.extend(registry.models.keys())
+        else:
+            for model_prefix in PREFIXES:
+                models.extend(
+                    [m for m in registry.models.keys() if m.startswith(model_prefix)]
+                )
+        models = sorted(models)
+        metafunc.parametrize("model_name", models, ids=models)
+
+
+@pytest.mark.slow
+@pytest.mark.external
+@pytest.mark.regression
+def test_model(
+    registry,
+    model_name,
+    tmp_path,
+    markers,
+    targets,
+    function_tmpdir,
+    original_regression,
+):
+    exclude = any(s in model_name for s in EXCLUDE)
+    devonly = "dev" in model_name and "not developmode" in markers
```

Comment:
> including also the concept of dev feature

@mjreno we talked offline, just to note for passersby, there is a mechanism here for test models which should run only in develop mode. I think the convention in the testmodels repo is to end the model name with "_dev".